### DISCUSSION TITLE
Embedding DAB into existing projects

### DISCUSSION THREAD

**Discussion by atrauzzi** [Azure/data-api-builder#2353] [Ideas]
I suppose this is starting out as a question, but based on my cursory searching, is likely an idea...

Does DAB have any potential to be embedded inside of an existing ASP.NET Core application?  For example, what if I'm already using HotChocolate for a GraphQL API so that I can write my own custom logic for mutations and do push.  But now what if I would like to leverage DAB for all the pesky CRUD?

By extension, could there be some synergies with the rest of the .NET ecosystem like Entity Framework to help DAB be more out-of-the-box configurable?

**Comment by seantleonard** [comment]
> Does DAB have any potential to be embedded inside of an existing ASP.NET Core application? 

Currently DAB isn't supported as an embeddable SDK within projects because changes we make would break dependent projects. This is still a good idea though that could be further explored. 


> By extension, could there be some synergies with the rest of the .NET ecosystem like Entity Framework to help DAB be more out-of-the-box configurable?

Do you have any specific examples of integrations or DAB customization you think would be really helpful here that we could talk through?

**Reply by atrauzzi** [reply]
Sorry for the long delay!  I think I laid out the scenario originally which is just to be able to have DAB be something inside of my existing asp.net app.

```csharp
app.AddOtherThings();
app.AddDab();
```

Being able to use DAB to offset CRUD could be interesting.
